x-pack/plugins/apm/common/critical_path/get_aggregated_critical_path_root_nodes.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { sumBy } from 'lodash';
+import type { CriticalPathResponse } from '../../server/routes/traces/get_aggregated_critical_path';
+
+export interface CriticalPathTreeNode {
+  nodeId: string;
+  children: CriticalPathTreeNode[];
+  countInclusive: number;
+  countExclusive: number;
+}
+
+export function getAggregatedCriticalPathRootNodes(params: {
```

Comment:
A unit test for this would help future interactions at this code.

Reply:
It's used in the API tests!